proof/courier.go

```diff
@@ -513,15 +514,16 @@ type Recipient struct {
 	Amount uint64
 }
 
-// HashMailCourierCfg is the config for the hashmail proof courier.
-type HashMailCourierCfg struct {
-	// ReceiverAckTimeout is the maximum time we'll wait for the receiver to
-	// acknowledge the proof.
-	ReceiverAckTimeout time.Duration `long:"receiveracktimeout" description:"The maximum time to wait for the receiver to acknowledge the proof."`
+// BackoffExecError is an error returned when the backoff execution fails.
+// This error wraps the underlying error returned by the execution function.
+// It allows the porter to determine whether the state machine should be halted
+// or not.
+type BackoffExecError struct {
+	execErr error
+}
 
-	// BackoffCfg configures the behaviour of the proof delivery
-	// functionality.
-	BackoffCfg *BackoffCfg
+func (e *BackoffExecError) Error() string {
+	return fmt.Sprintf("backoff exec error: %s", e.execErr.Error())
```

Comment:
Do we need a `nil` check on the `execErr` here? Or can we be reasonably sure it is always set?

Reply:
AFAICT `BackoffHandle.Exec()` is the only place we're building such an error. But a nil check would be good anyways, could just return "backoff exec error" imo.